notifications.bs

```diff
@@ -1098,6 +1104,10 @@ method steps are:
 
    <li><p>Run the <a for=/>notification show steps</a> for <var>notification</var>.
 
+   <li><p>Set <var>notification</var>'s <a for=notification>service worker registration</a>'s
+   <a for="service worker registration">has <code>showNotification()</code> been successfully invoked</a>
+   to true.
```

Comment:
Nothing sets it back to false, perhaps the push event handler should set it false in https://github.com/w3c/push-api/pull/385?

Reply:
I also wonder what would happen if another push event is fired before the previous one finishes, because then this value would conflict. Perhaps we could have PushEvent.showNotification() (where the default arg is the DWP parse result) to make it per-event? 🤔

Reply:
Right, I was waiting to see if this approach (which also means you have to process pushes sequentially) works for everyone, but then nobody gave feedback for what feels like an eternity…

Reply:
@saschanaz it seems good to resolve this as well.

The idea here is indeed for the Push API PR to set this to false and for the Push API specification to enforce incoming messages are processed sequentially. (Perhaps scoped to a site, but I don't think we have to go into that much detail for now.)

Reply:
I don't think Gecko waits for service worker until processing the next event, the [PushService notifies the service worker](https://searchfox.org/mozilla-central/rev/67c9592293e8d9f143e6053c8cfd6b7723207e3c/dom/push/PushService.sys.mjs#685-698) and moves on for the next push message. There should be something that ties a PushEvent and showNotification.

Reply:
Does WebKit makes sure it's sequential btw? [queueTaskToFirePushEvent](https://searchfox.org/wubkat/rev/c46918d1cba365017d1df98e3d14f04db42320bc/Source/WebCore/workers/service/context/ServiceWorkerThread.cpp#290-317) looks like the queued task would end after adding a callback to the extended lifetime promise, is there an internal test for that?

Reply:
I think the idea is to not queue another task until the callback is called.